src/Umbraco.Web.Common/Media/ImageSharpImageUrlTokenGenerator.cs

```diff
@@ -0,0 +1,89 @@
+    {
+        private readonly byte[]? _hmacSecretKey;
+        private readonly CaseHandlingUriBuilder.CaseHandling _caseHandling;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageSharpImageUrlTokenGenerator" /> class.
+        /// </summary>
+        /// <param name="imagingSettings">The Umbraco imaging settings.</param>
+        public ImageSharpImageUrlTokenGenerator(IOptions<ImagingSettings> imagingSettings)
+            : this(imagingSettings.Value.HMACSecretKey, CaseHandlingUriBuilder.CaseHandling.LowerInvariant)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageSharpImageUrlTokenGenerator" /> class.
+        /// </summary>
+        /// <param name="hmacSecretKey">The HMAC security key.</param>
+        /// <remarks>
+        /// This constructor is only used for testing.
+        /// </remarks>
+        internal ImageSharpImageUrlTokenGenerator(byte[]? hmacSecretKey)
+            : this(hmacSecretKey, CaseHandlingUriBuilder.CaseHandling.LowerInvariant)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageSharpImageUrlTokenGenerator" /> class.
+        /// </summary>
+        /// <param name="hmacSecretKey">The HMAC security key.</param>
+        /// <param name="caseHandling">Determines case handling for the result.</param>
+        protected ImageSharpImageUrlTokenGenerator(byte[]? hmacSecretKey, CaseHandlingUriBuilder.CaseHandling caseHandling)
+        {
+            _hmacSecretKey = hmacSecretKey;
+            _caseHandling = caseHandling;
+        }
+
+        /// <inheritdoc />
+        public string? GetImageUrlToken(string imageUrl, IEnumerable<KeyValuePair<string, string?>> commands)
+        {
+            if (_hmacSecretKey == null || _hmacSecretKey.Length == 0)
+            {
+                return null;
+            }
+
+            QueryString queryString = CreateImageUrlTokenQueryString(commands);
+            string value = CaseHandlingUriBuilder.BuildRelative(_caseHandling, null, imageUrl, queryString);
+
+            return ComputeImageUrlToken(value, _hmacSecretKey);
+        }
+
+        /// <summary>
+        /// Creates the query string that is included in the image URL token.
+        /// </summary>
+        /// <param name="commands">The commands to include.</param>
+        /// <returns>
+        /// The query string.
+        /// </returns>
+        /// <remarks>
+        /// The ImageSharp middleware computes the token using only known commands,
+        /// so if you've removed default processors or add unknown commands using the FurtherOptions,
+        /// you can use this method to filter out any unsupported/unknown commands.
+        /// </remarks>
+        protected virtual QueryString CreateImageUrlTokenQueryString(IEnumerable<KeyValuePair<string, string?>> commands)
+            => QueryString.Create(commands);
```

Comment:
This is now fixed by using `Lazy<IEnumerable<IImageWebProcessor>>` and fetching the known commands the first time it's needed (removing the cyclic dependency in the constructor), so there's no need to manually remove unsupported commands.

Because ImageSharp doesn't support skipping token validation based on specific conditions (see https://github.com/SixLabors/ImageSharp.Web/pull/265#issuecomment-1131309819), whitelisting of existing commands doesn't work and I've therefore removed this virtual method.